crates/ide-assists/src/handlers/extract_function.rs

```diff
@@ -1318,7 +1318,14 @@ fn make_call(ctx: &AssistContext<'_>, fun: &Function, indent: IndentLevel) -> St
     } else if fun.ret_ty.is_unit() && (!fun.outliving_locals.is_empty() || !expr.is_block_like()) {
         buf.push(';');
     }
-    buf
+
+    // The replacement text should be interpreted as-is, as it is expected to be
+    // a normal Rust expression. We escape '\' and '$' so they don't get
+    // treated as snippet-specific constructs.
+    //
+    // Note that we don't need to escape the other characters that can be escaped,
+    // because they wouldn't be treated as snippet-specific constructs without '$'.
+    buf.replace('\\', "\\\\").replace('$', "\\$")
```

Comment:
Your test is failing because this function (`make_call()`) is generating the call to the extracted function, not the extracted function body (which is separately generated by `format_function()`). While we *do* need to escape the call arguments in this function, I'd expect another test for that.